SIT.Manager.Avalonia/ViewModels/Installation/InstallViewModel.cs

```diff
@@ -0,0 +1,92 @@
+﻿using CommunityToolkit.Mvvm.ComponentModel;
+using CommunityToolkit.Mvvm.Messaging;
+using SIT.Manager.Avalonia.Interfaces;
+using SIT.Manager.Avalonia.Models.Installation;
+using System;
+using System.Threading.Tasks;
+
+namespace SIT.Manager.Avalonia.ViewModels.Installation;
+
+public partial class InstallViewModel : InstallationViewModelBase
+{
+    private readonly IInstallerService _installerService;
+
+    private readonly Progress<double> _downloadProgress = new();
+    private readonly Progress<double> _extractionProgress = new();
+
+    [ObservableProperty]
+    private double _downloadProgressPercentage = 0;
+
+    [ObservableProperty]
+    private double _extractionProgressPercentage = 0;
+
+    [ObservableProperty]
+    private double _installProgressPercentage = 0;
+
+    public InstallViewModel(IInstallerService installerService) : base()
+    {
+        _installerService = installerService;
+
+        _downloadProgress.ProgressChanged += DownloadProgress_ProgressChanged;
+        _extractionProgress.ProgressChanged += ExtractionProgress_ProgressChanged;
+    }
+
+    private void DownloadProgress_ProgressChanged(object? sender, double e)
+    {
+        // For some reason this is 100 times smaller than expected so just x100 I guess?
+        DownloadProgressPercentage = e * 100;
+        UpdateInstallProgress();
+    }
+
+    private void ExtractionProgress_ProgressChanged(object? sender, double e)
+    {
+        ExtractionProgressPercentage = e;
+        UpdateInstallProgress();
+    }
+
+    private void UpdateInstallProgress()
+    {
+        InstallProgressPercentage = (DownloadProgressPercentage + ExtractionProgressPercentage) / 2;
+    }
+
+    private async Task RunInstaller()
+    {
+        DownloadProgressPercentage = 0;
+        ExtractionProgressPercentage = 0;
+
+        try
+        {
+            if (IsServerInstall)
+            {
+                await _installerService.InstallServer(CurrentInstallProcessState.RequestedVersion, CurrentInstallProcessState.SptAkiInstallPath, _downloadProgress, _extractionProgress);
+            }
+            else if (IsSitInstall)
+            {
+                await _installerService.InstallSit(CurrentInstallProcessState.RequestedVersion, CurrentInstallProcessState.EftInstallPath, _downloadProgress, _extractionProgress);
```

Comment:
Given how similar these two are is it at all possible we could combine these into one method and pass an identifying enum or something?

Reply:
Possibly... I'd leave that for the future though so that we can get them merged together nicely with whatever shared things they actually do.